bot/exts/info/subscribe.py

```diff
@@ -178,22 +233,73 @@ async def cog_load(self) -> None:
-            row = index // ITEMS_PER_ROW
-            button_view.add_item(SingleRoleButton(role, role.role_id in author_roles, row))
+        view = prepare_self_assignable_roles_view(ctx, self.assignable_roles)
 
-        await ctx.send(
+        message = await ctx.send(
             "Click the buttons below to add or remove your roles!",
-            view=button_view,
-            delete_after=DELETE_MESSAGE_AFTER,
+            view=view,
         )
+        # Keep reference of the message that contains the view to be deleted
+        view.anchor_message = message
+
+    async def __search_for_self_assignable_roles_message(self) -> discord.Message:
+        """
+        Searches for the message that holds the self assignable roles view.
+
+        If the initial message isn't found, a new one will be created.
+        This message will always be needed to attach the persistent view to it
+        """
+        roles_channel: discord.TextChannel = await get_or_fetch_channel(constants.Channels.roles)
+
+        async for message in roles_channel.history(limit=30):
+            if message.content == self.SELF_ASSIGNABLE_ROLES_MESSAGE:
+                log.debug(f"Found self assignable roles view message: {message.id}")
+                return message
+
+        log.debug("Self assignable roles view message hasn't been found, creating a new one.")
+        view = AllSelfAssignableRolesView()
+        view.add_item(ShowAllSelfAssignableRolesButton(self.assignable_roles))
+        return await roles_channel.send(self.SELF_ASSIGNABLE_ROLES_MESSAGE, view=view)
+
+    def __attach_persistent_roles_view(
+            self,
+            placeholder_message: discord.Message
+    ) -> None:
+        """
+        Attaches the persistent view that toggles self assignable roles to its placeholder message.
+
+        The message is searched for/created upon loading the Cog.
+
+        Parameters
+        __________
+            :param placeholder_message: The message that will hold the persistent view allowing
+            users to toggle the RoleButtonView
+        """
+        view = AllSelfAssignableRolesView()
+        view.add_item(ShowAllSelfAssignableRolesButton(self.assignable_roles))
+        self.bot.add_view(view, message_id=placeholder_message.id)
+
+
+def prepare_self_assignable_roles_view(
+        trigger_action: commands.Context | Interaction,
+        assignable_roles: list[AssignableRole]
+) -> discord.ui.View:
+    """Prepares the view containing the self assignable roles before its sent."""
+    author = trigger_action.author if isinstance(trigger_action, commands.Context) else trigger_action.user
+    author_roles = [role.id for role in author.roles]
+    button_view = RoleButtonView(member=author)
+    button_view.anchor_message = trigger_action.message
+
+    for index, role in enumerate(assignable_roles):
+        row = index // ITEMS_PER_ROW
+        button_view.add_item(SingleRoleButton(role, role.role_id in author_roles, row))
```

Comment:
So I know that this is copied from existing code, but I think that the button creation should be done inside `RoleButtonView`'s init. It already takes the author, and then it can make the buttons for itself. You then also don't need this function at all.

Reply:
Good eye !  
This has been done here 37b43a6c04af0498d96bd28ec544973d2e653941